### ISSUE TITLE
Got only error message

### ISSUE THREAD

**Issue by unmenschlich** [citronalco/calibre-dnb#6]
i have a SRU token and every time i test it with a random ebook, i get message: ERROR: Got no valid response.
With browser i get this and downloading cover works too

```
1.11MARC21-xmlxml 00000nam a2200000uc 4500 1068814470 DE-101 20171202083524.0 cr|||||||||||| 150324s2015 gw |||||o|||| 00||||ger 15,O04 dnb DE-101 1068814470 9783842015067 978-3-8420-1506-7 9783842015067 urn urn:nbn:de:101:1-201503246928 (DE-599)DNB1068814470 (OCoLC)906205815 1240 ger DE-101 9999 ger jpn XA-DE-HH 2\p 300 DE-101 23sdnb 1\p Sugisaki, Yukiru Verfasser aut 1001 Knights 02 Yukiru Sugisaki Hamburg Tokyopop Verlag 2015 Online-Ressource Text txt rdacontent Computermedien c rdamedia Online-Ressource cr rdacarrier 1001 Knights 2 Lizenzpflichtig Archivierung/Langzeitarchivierung gewährleistet DE-101 pdager (Produktform)Electronic book text (BISAC Subject Heading)CGN004050 (BISAC Subject Heading)CGN004050 Detektiv Detektei Mystery Übernatürliches mysteriös Dschinn Abendland Märchen Prinzessin Siegel Agent Agenten Bruder Brüder Zwillinge Naito Yuta (VLB-WN)9182 Druckausg. 9783842010758 1001 Knights 2 DE-101a DE-101b https://nbn-resolving.org/urn:nbn:de:101:1-201503246928 Resolving-System https://d-nb.info/1068814470/34 Langzeitarchivierung Nationalbibliothek application/pdf http://www.tokyopop.de Verlag 1\p npi 20200611 DE-101 https://d-nb.info/provenance/plan#npi 2\p dnb 0,907 20150325 DE-101 https://d-nb.info/provenance/plan#dnb ro ra XAM 93 1.0 121.1num=9783842015067 NOT (mat=film OR mat=music OR mat=microfiches OR cod=tt)100MARC21-xmlXXXXXX
```

Job result
```
Querying: tit="1001 Knights 2" AND per="Yukiru Sugisaki" AND num="9783842015067" NOT (mat=film OR mat=music OR mat=microfiches OR cod=tt)
Query URL: https://services.dnb.de/sru/dnb?version=1.1&accessToken=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX&maximumRecords=100&operation=searchRetrieve&recordSchema=MARC21-xml&query=tit%3D%221001%20Knights%202%22%20AND%20per%3D%22Yukiru%20Sugisaki%22%20AND%20num%3D%229783842015067%22%20NOT%20%28mat%3Dfilm%20OR%20mat%3Dmusic%20OR%20mat%3Dmicrofiches%20OR%20cod%3Dtt%29
ERROR: Got no valid response.
per="1001 Knights 2" AND tit="Yukiru Sugisaki" AND num="9783842015067" NOT (mat=film OR mat=music OR mat=microfiches OR cod=tt)
Querying: per="1001 Knights 2" AND tit="Yukiru Sugisaki" AND num="9783842015067" NOT (mat=film OR mat=music OR mat=microfiches OR cod=tt)
Query URL: https://services.dnb.de/sru/dnb?version=1.1&accessToken=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX&maximumRecords=100&operation=searchRetrieve&recordSchema=MARC21-xml&query=per%3D%221001%20Knights%202%22%20AND%20tit%3D%22Yukiru%20Sugisaki%22%20AND%20num%3D%229783842015067%22%20NOT%20%28mat%3Dfilm%20OR%20mat%3Dmusic%20OR%20mat%3Dmicrofiches%20OR%20cod%3Dtt%29
ERROR: Got no valid response.
"1001 Knights 2" AND "Yukiru Sugisaki" AND num="9783842015067" NOT (mat=film OR mat=music OR mat=microfiches OR cod=tt)
Querying: "1001 Knights 2" AND "Yukiru Sugisaki" AND num="9783842015067" NOT (mat=film OR mat=music OR mat=microfiches OR cod=tt)
Query URL: https://services.dnb.de/sru/dnb?version=1.1&accessToken=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX&maximumRecords=100&operation=searchRetrieve&recordSchema=MARC21-xml&query=%221001%20Knights%202%22%20AND%20%22Yukiru%20Sugisaki%22%20AND%20num%3D%229783842015067%22%20NOT%20%28mat%3Dfilm%20OR%20mat%3Dmusic%20OR%20mat%3Dmicrofiches%20OR%20cod%3Dtt%29
ERROR: Got no valid response.
num=9783842015067 NOT (mat=film OR mat=music OR mat=microfiches OR cod=tt)
Querying: num=9783842015067 NOT (mat=film OR mat=music OR mat=microfiches OR cod=tt)
Query URL: https://services.dnb.de/sru/dnb?version=1.1&accessToken=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX&maximumRecords=100&operation=searchRetrieve&recordSchema=MARC21-xml&query=num%3D9783842015067%20NOT%20%28mat%3Dfilm%20OR%20mat%3Dmusic%20OR%20mat%3Dmicrofiches%20OR%20cod%3Dtt%29
ERROR: Got no valid response. 

******************************************************************************** 
The identify phase took 1.21 seconds 
The longest time (1.067534) was taken by: DNB_DE 
Merging results from different sources 
We have 0 merged results, merging took: 0.00 seconds 
Failed to download metadata for 1001 Knights 2 

****************************** DNB_DE Covers ****************************** 
Request extra headers: [('User-agent', 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:84.0) Gecko/20100101 Firefox/84.0')] 
Failed to download valid cover 
Took 0.24401330947875977 seconds 
Downloading cover from: https://portal.dnb.de/opac/mvb/cover.htm?isbn=9783842015067
Could not download Cover 

********************************************************************************

```

**Comment by citronalco** [comment]
Looks like there's an issue accessing the internet: Even downloading the cover image fails.

Does internet access work in other Calibre plugins, i.e. with the Google metadata plugin? 
If not, and if you're behind a proxy, this might be helpful: https://manual.calibre-ebook.com/faq.html#how-do-i-get-calibre-to-use-my-http-proxy

**Comment by unmenschlich** [comment]
get Metadata from Google, Amazon.de works
calibre.exe and calibre-parallel.exe have access to internet and see with SysInternals Procmon that calibre-parallel send and receive data from httpd-e02.dnb.de

**Comment by citronalco** [comment]
Are you using a proxy?

**Comment by unmenschlich** [comment]
no, only a vpn directly from router but nothing on my computer

**Comment by unmenschlich** [comment]
i change on plugin https to http urls to capture traffic from plugin but looks like your plugin works for me with http.
but with other plugins https works
job output
```

#################### Log for 1001 Knights 2 ####################
Running identify query with parameters: 
{'title': '1001 Knights 2', 'authors': ['Yukiru Sugisaki'], 'identifiers': {'isbn': '9783842015067'}, 'timeout': 30} 
Using plugins: DNB_DE (3, 0, 0) 
The log from individual plugins is below 

****************************** DNB_DE (3, 0, 0) ****************************** 
Found 1 results 
Downloading from DNB_DE took 0.8688232898712158 


--- 
Title               : 1001 Knights 02
Title sort          : 1001 Knights 02
Author(s)           : Sugisaki, Yukiru [Sugisaki, Yukiru]
Publisher           : Hamburg ; Tokyopop Verlag
Series              : 1001 Knights #2
Languages           : ger
Published           : 2015-01-01T11:30:00+00:00
Identifiers         : isbn:9783842015067, urn:urn:nbn:de:101:1-201503246928, dnb-idn:1068814470, ddc:300 
tit="1001 Knights 2" AND per="Yukiru Sugisaki" AND num="9783842015067" NOT (mat=film OR mat=music OR mat=microfiches OR cod=tt)
Querying: tit="1001 Knights 2" AND per="Yukiru Sugisaki" AND num="9783842015067" NOT (mat=film OR mat=music OR mat=microfiches OR cod=tt)
Query URL: http://services.dnb.de/sru/dnb?version=1.1&accessToken=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX&maximumRecords=100&operation=searchRetrieve&recordSchema=MARC21-xml&query=tit%3D%221001%20Knights%202%22%20AND%20per%3D%22Yukiru%20Sugisaki%22%20AND%20num%3D%229783842015067%22%20NOT%20%28mat%3Dfilm%20OR%20mat%3Dmusic%20OR%20mat%3Dmicrofiches%20OR%20cod%3Dtt%29
Got records: 0
per="1001 Knights 2" AND tit="Yukiru Sugisaki" AND num="9783842015067" NOT (mat=film OR mat=music OR mat=microfiches OR cod=tt)
Querying: per="1001 Knights 2" AND tit="Yukiru Sugisaki" AND num="9783842015067" NOT (mat=film OR mat=music OR mat=microfiches OR cod=tt)
Query URL: http://services.dnb.de/sru/dnb?version=1.1&accessToken=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX&maximumRecords=100&operation=searchRetrieve&recordSchema=MARC21-xml&query=per%3D%221001%20Knights%202%22%20AND%20tit%3D%22Yukiru%20Sugisaki%22%20AND%20num%3D%229783842015067%22%20NOT%20%28mat%3Dfilm%20OR%20mat%3Dmusic%20OR%20mat%3Dmicrofiches%20OR%20cod%3Dtt%29
Got records: 0
"1001 Knights 2" AND "Yukiru Sugisaki" AND num="9783842015067" NOT (mat=film OR mat=music OR mat=microfiches OR cod=tt)
Querying: "1001 Knights 2" AND "Yukiru Sugisaki" AND num="9783842015067" NOT (mat=film OR mat=music OR mat=microfiches OR cod=tt)
Query URL: http://services.dnb.de/sru/dnb?version=1.1&accessToken=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX&maximumRecords=100&operation=searchRetrieve&recordSchema=MARC21-xml&query=%221001%20Knights%202%22%20AND%20%22Yukiru%20Sugisaki%22%20AND%20num%3D%229783842015067%22%20NOT%20%28mat%3Dfilm%20OR%20mat%3Dmusic%20OR%20mat%3Dmicrofiches%20OR%20cod%3Dtt%29
Got records: 0
num=9783842015067 NOT (mat=film OR mat=music OR mat=microfiches OR cod=tt)
Querying: num=9783842015067 NOT (mat=film OR mat=music OR mat=microfiches OR cod=tt)
Query URL: http://services.dnb.de/sru/dnb?version=1.1&accessToken=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX&maximumRecords=100&operation=searchRetrieve&recordSchema=MARC21-xml&query=num%3D9783842015067%20NOT%20%28mat%3Dfilm%20OR%20mat%3Dmusic%20OR%20mat%3Dmicrofiches%20OR%20cod%3Dtt%29
Got records: 1
Parsing records
Extracted Publisher: Tokyopop Verlag
Extracted Publisher Location: Hamburg
Extracted Publication Year: 2015-01-01 12:30:00
Extracted Title: 1001 Knights 02
Extracted Title_Sort: 1001 Knights 02
Extracted Authors: Sugisaki, Yukiru
Extracted Author_Sort: Sugisaki, Yukiru
Extracted ID IDN: 1068814470
Extracted ID URN: urn:nbn:de:101:1-201503246928
Extracted ID ISBN: 9783842015067
Extracted ID DDC: 300
Extracted Series Index from Field 490: 2
Extracted Series from Field 490: 1001 Knights
Extracted non-GND Subjects: Detektiv Detektei Mystery Übernatürliches mysteriös Dschinn Abendland Märchen Prinzessin Siegel Agent Agenten Bruder Brüder Zwillinge Naito Yuta
Extracted Languages: ger
Final formatted result: 
Title               : 1001 Knights 02
Title sort          : 1001 Knights 02
Author(s)           : Sugisaki, Yukiru [Sugisaki, Yukiru]
Publisher           : Hamburg ; Tokyopop Verlag
Series              : 1001 Knights #2
Languages           : ger
Published           : 2015-01-01T11:30:00+00:00
Identifiers         : isbn:9783842015067, urn:urn:nbn:de:101:1-201503246928, dnb-idn:1068814470, ddc:300 

******************************************************************************** 
The identify phase took 1.02 seconds 
The longest time (0.868823) was taken by: DNB_DE 
Merging results from different sources 
We have 1 merged results, merging took: 0.00 seconds 

****************************** DNB_DE Covers ****************************** 
Request extra headers: [('User-agent', 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.88 Safari/537.36')] 
Failed to download valid cover 
Took 0.23355340957641602 seconds 
Downloading cover from: http://portal.dnb.de/opac/mvb/cover.htm?isbn=9783842015067
Could not download Cover 

******************************************************************************** 
```

captured traffic why not downloading cover too
```
GET http://portal.dnb.de/opac/mvb/cover.htm?isbn=9783842015067 HTTP/1.1
Accept-Encoding: identity
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.88 Safari/537.36
Host: portal.dnb.de
Connection: close

HTTP/1.1 302 Found
Date: Tue, 02 Feb 2021 17:13:16 GMT
Server: Apache
Location: https://portal.dnb.de/opac/mvb/cover.htm?isbn=9783842015067
Content-Length: 306
Connection: close
Content-Type: text/html; charset=iso-8859-1

<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>302 Found</title>
</head><body>
<h1>Found</h1>
<p>The document has moved <a href="https://portal.dnb.de/opac/mvb/cover.htm?isbn=9783842015067">here</a>.</p>
<hr>
<address>Apache Server at portal.dnb.de Port 80</address>
</body></html>
```

**Comment by citronalco** [comment]
I tried to reproduce your issue and installed a Calibre 5.10.1 (64 Bit) on a clean Windows 10 machine, added v.3.0.0 of the DNB plugin from Github's "Releases" page and tried to download metadata for a book. Works. Tried to download some covers. Works. I gave the book the same title, author and ISBN as in your example so that the query string was identical to yours. Works. No problems with https.

As most people don't complain if something does not work, I'm pretty sure some others have your issue too. So I want to get that sorted out.
Can you tell me the exact Calibre and DNB plugin version you're using?

**Comment by unmenschlich** [comment]
thank you for help,
i use calibre Portable v5.10.1 (32 Bit) from here https://calibre-ebook.com/download_portable
dnb plugin v3.0.0 from calibre own plugin downloader, at the moment with http on SRU urls to get metadatas

i dont know about python or calibre plugin specification but is it possibly that i get more verbose from https request?

**Comment by citronalco** [comment]
I'm still irritated that https works for you with other plugins.
Basically all metadata plugins use the "browser" object that comes from Calibre to interact with web servers.
The DNB plugin also uses this object, and calls the same "browser.open_novisit" method as other plugins, i.e. the Google plugin. It's quite simple. No differences here.

The easiest way to do some debugging is simply to write some "log.info('variable foo has value ' + foo)" statements in the code.

You've written that you're using a VPN. Does it work without VPN?

**Comment by unmenschlich** [comment]
i test it now with usb tethering from smartphone, ethernet and wlan adapter was disconnected but i got only error message.
other test was normally install calibre 64bit, download plugin but dont work.
vpn server with a german ip dont work too.
with firefox same urls are working.

maybe its just me, i dont know why.
because debugging, if you can say me how i would do it

**Comment by citronalco** [comment]
Downloaded Calibre, this time 32 bit, installed on a rather fresh Win10 installation, added the "Deutsche Nationalbibliothek" addon directly within Calibre from the "plugins" list, restarted Calibre, changed the included manual's title to "1001 Knights 2", author to "Yukiru Sugisaki" and ISBN to "9783842015067", set the metadata source to DNB and downloaded metadata. Works.

Sorry, I still can't reproduce the issue. No clue.

**Comment by unmenschlich** [comment]
its ok, i have no ideas why it not works.

**Comment by citronalco** [comment]
If you ever find out what it was: Let me know!

**Comment by citronalco** [comment]
Cannot reproduce

**Comment by citronalco** [comment]
Could be the same issue as https://github.com/citronalco/calibre-dnb/issues/9

**Comment by citronalco** [comment]
Fixed in v3.1.7 released some seconds ago.
